.github/workflows/static-code-analysis.yml

```diff
@@ -0,0 +1,51 @@
+name: "Static code analysis"
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  schedule:
+    # Everyday at midnight
+    - cron: '0  0 * * *'
+
+jobs:
+  codeql:
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: >
+           sudo apt update &&
+           sudo apt install -y autoconf automake autopoint check cifs-utils
+           clang dh-apparmor dnsutils docbook-xml docbook-xsl gettext
+           krb5-config libc-ares-dev libcmocka-dev libcollection-dev
+           libdbus-1-dev libdhash-dev libglib2.0-dev libini-config-dev
+           libkeyutils-dev libkrb5-dev libldap2-dev libldb-dev libltdl-dev
+           libnfsidmap-dev libnl-3-dev libnl-route-3-dev libpam0g-dev
+           libpcre2-dev libpopt-dev libsasl2-dev libselinux1-dev
+           libsemanage1-dev libsmbclient-dev libsystemd-dev libtalloc-dev
+           libtdb-dev libtevent-dev libtool libtool-bin libxml2-utils make
+           python-dev python3-dev samba-dev systemd xml-core xsltproc
+           libssl-dev fakeroot faketime libnss-wrapper libuid-wrapper
+           libpam-wrapper python-pytest python3-ldap python3-ldb
+           python3-requests python-psutil ldap-utils slapd systemtap-sdt-dev
+           libhttp-parser-dev libcurl4-openssl-dev krb5-kdc
+           krb5-admin-server krb5-user uuid-dev dbus python-dbus pep8
+           libssl-dev gnutls-bin softhsm2 libp11-kit-dev libunistring-dev
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+
+      - name: Compile sssd
+        run: |
+           source contrib/fedora/bashrc_sssd
+           reconfig --with-smb-idmap-interface-version=5
+           PROCESSORS=$(/usr/bin/getconf _NPROCESSORS_ONLN)
+           make -j$PROCESSORS
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v1
```

Comment:
You can also enable python and more thorough rules E.g. https://github.com/authselect/authselect/blob/master/.github/workflows/analyze.yml#L28

Reply:
I'm hesitant about enabling this option. Without using it the tools reports ~30 issues, I decided to fix only 4 of them because the rest weren't worth the effort (mostly related with sensitive information in tests). If I use that option then the tool reports more than 1000 issues, and only 15 or so are worth fixing. I think that enabling it will generate too much noise. Do you have a strong opinion on this matter?

Reply:
Once it runs on master it will only report newly added issues on pull requests, so there is no harm in enabling it. We can mass-dismiss what is not worth fixing.

The goal is to avoid introducing any new issues and increase code quality, which can be done easily with diff scan on pull requests, but author can of course dismiss false positives/intentional issues. If we find out that we dismiss majority of reports then we can disable it later.

My small experience with codeql in authselect is that these additional rules report not really important issues but usually valid and makes sense to fix them in pull request.

Reply:
Agreed.